spec-0007/index.md

```diff
@@ -0,0 +1,149 @@
+
+Two strong motivations for moving over to `Generator`s are:
+
+1. they avoid naïve seeding strategies, such as using successive integers, via the underlying [SeedSequence](https://numpy.org/doc/stable/reference/random/parallel.html#seedsequence-spawning);
+2. they avoid using global state (from `numpy.random.mtrand._rand`).
+
+Our recommendation here is a deprecation strategy which does not in _all_ cases adhere to the Hinsen principle[^hinsen],
+although it could very nearly do so by enforcing the use of `rng` as a keyword argument.
+
+[^hinsen]: The Hinsen principle states, loosely, that code should, whether executed now or in the future, return the same result, or raise an error.
+
+The [deprecation strategy](https://github.com/scientific-python/specs/pull/180#issuecomment-1515248009) is:
+
+1. Accept both `rng` and `random_state`/`seed` keyword arguments.
+2. If both are specified, raise an error.
+3. If neither is specified and `np.random.seed` has been used to set the seed, emit a `FutureWarning` about the upcoming change in behavior.
+4. If `random_state`/`seed` is passed by keyword or by position, treat it as before, but:
+
+   - Emit a `DeprecationWarning` if passed by keyword, warning about the deprecation of keyword `random_state` in favor of `rng`.
+   - Emit a `FutureWarning` if passed by position, warning about the change in behavior of the positional argument.
+
+5. If `rng` is passed by keyword, standardize it using `numpy.random.default_rng`.
+6. After the deprecation period, use only `rng`, validated with `numpy.random.default_rng`.
+   Raise an error if `random_state`/`seed` is provided.
+
+   By now, the function signature, with type annotations, could look like this:
+
+   ```python
+   from collections.abc import Sequence
+   import numpy as np
+
+
+   SeedLike = int | np.integer | Sequence[int] | np.random.SeedSequence
+   RNGLike = np.random.Generator | np.random.BitGenerator
+
+
+   def my_func(rng: RNGLike | SeedLike | None = None):
+       """My function summary.
+
+       Parameters
+       ----------
+       rng : `numpy.random.Generator`, optional
+           Pseudorandom number generator state. When `rng` is None, a new
+           `numpy.random.Generator` is created using entropy from the
+           operating system. Types other than `numpy.random.Generator` are
+           passed to `numpy.random.default_rng` to instantiate a `Generator`.
+       """
+       rng = np.random.default_rng(rng)
+
+       ...
+
+   ```
+
+   Also note the suggested language for the `rng` parameter docstring, which encourages the user to pass a `Generator` or None, but allows for other types accepted by `numpy.random.default_rng` (captured by the type annotation).
+
+### Impact
+
+There are three classes of users, which will be affected to varying degrees.
+
+1. Those who do not attempt to control the random state.
+   Their code will immediately switch from using the unseeded global `RandomState` to using an unseeded `Generator`.
```

Comment:
```suggestion
   Their code will switch from using the unseeded global `RandomState` to using an unseeded `Generator`.
```
There's not really a reason that it needs to be immediate, and that's not what the decorator does now.